src/array.rs

```diff
@@ -1000,20 +997,8 @@ impl<T: Element> PyArray<T, Ix1> {
     /// });
     /// ```
     pub fn from_exact_iter(py: Python<'_>, iter: impl ExactSizeIterator<Item = T>) -> &Self {
-        // NumPy will always zero-initialize object pointers,
-        // so the array can be dropped safely if the iterator panics.
-        unsafe {
-            let len = iter.len();
-            let array = Self::new(py, [len], false);
-            let mut idx = 0;
-            for item in iter {
-                assert!(idx < len);
-                array.uget_raw([idx]).write(item);
-                idx += 1;
-            }
-            assert!(idx == len);
-            array
-        }
+        let data = iter.collect::<Box<[_]>>();
```

Comment:
My initial aim was to avoid changing the API, but since `from_*iter` is probably not the most used API and we are still at version 0.x, this does not really seem to be appropriate.

As for using `Box<[_]>` instead of `Vec<_>`, I was thinking that the exact case will not end up with excess capacity and hence could go from vec to boxed slice without re-allocation. But this is not really useful as a boxed slice and a vec without excess capacity will basically end up creating the same `PySliceContainer` instance.

Hence, I'll a commit removing the `from_exact_iter` method as any specializations that the standard library provides with pass through the generic signature of `from_iter` in any case.